### DISCUSSION TITLE
How to work with toConstantDependency function when the second argument of that function includes a certain member imported with import statement

### DISCUSSION THREAD

**Discussion by liuliangsir** [webpack/webpack#15662] [Q&A]
As the title above says, how to resolve the specified use case? Here is a short(or simplified) example about more detail information. The followings are a bunch of code snippets.

First, I would like to achieve a certain feature, which is to replace some text during compilation stage via webpack plugin.

```js
import time from 'time'

// replace some patterned templates like `__(foo)` with some translations
// in the following example, `__('Time is ' + time)` is replaced with `'время ' + time` during compilation stage
__('Time is ' + time)
```

Here is a certain code snippet about webpack plugin
```js
// replacement is a string like 'time' but the raw format of replacement argument is a variable imported from other module using import statement
toConstantDependency(parser, replacement)(callExpressionNode)
```

After run some bootstrap script like 'npm start', The compilation process is pass and generate a lot of assets. But the web app is not working. After debugging, I think that `toConstantDependency` seems to be a root cause as it cannot work well with imported module using import statement. Finally, I find a workaround that is using a temporary variable to take imported variable like the following.

```js
import oldTime from 'time'

const time = oldTime 

// replace some patterned templates like `__(foo)` with some translations
// in the following example, `__('Time is ' + time)` is replaced with `'время ' + time` during compilation stage
__('Time is ' + time)
```

**Comment by vankop** [comment]
you should not touch import part (if I understand what you are trying to achieve ), e.g.
module code
```js
import time from "time";
toConst(time, otherArg);
```
and if you trying to replace `toConst(time, otherArg)` only replace non-import part. ( to parse import specifier just use `parser.walkExpression(argumentNode)` )

**Reply by liuliangsir** [reply]
@vankop cc I add more details

**Reply by vankop** [reply]
so you should replace CallExpression `callee` in your example this is `__` and part of argument ( `'Time is '`) but don't touch `time`

**Reply by liuliangsir** [reply]
@vankop is there any other solution? Because the resulting position of variable(in the above example, that is `time`) inside translation text is changing dynamically depending on language self context. Therefore, I need to allow for the variable instead of ignoring that when translating

**Comment by liuliangsir** [comment]
@privatenumber @sokra cc
